src/components/AddCard.tsx

```diff
@@ -1,42 +1,39 @@
 import React, {useState, useEffect} from 'react';
-import {connect} from "react-redux";
-import {getTodos, postTodos, showEditItem, saveEditItem, closePressed} from "../redux/actions/index"
+import {useSelector, useDispatch} from 'react-redux'
+import {getTodos, postTodos, showEditItem, closePressed} from "../redux2/actions/index"
+import { RootState} from "../redux2/index"
+// import {getTodos, postTodos, showEditItem, saveEditItem, closePressed} from "redux_path/actions/index"
+
 import Presentation from "./AddPresentation";
-import {TodoItem} from "./models"
+// import {TodoItem} from "src/components/models"
 
-function AddCard(props:any) {
-    const [todos, setTodos] = useState([]);
+function AddCard() {
+    const dispatch = useDispatch()
+    const todos = useSelector((state: RootState)  => state.todos)
+    const newItem = useSelector((state: RootState)  => state.postTodo)
     const [showForm, setShowForm] = useState(false);
     const [textData, setTextData] = useState("");
     const [editData, setEditData] = useState("");
 
     useEffect(()=>{
-        props.getTodos()
-        props.postTodos({id:2, text: "Typescript", showEdit: false})
-        props.postTodos({id:3, text: "Styled Component", showEdit: false})
+        dispatch(getTodos())
+        dispatch(postTodos("Typescript"))
+        dispatch(postTodos("Styled Component"))
     }, [])
 
     useEffect(()=>{
-        if(props.todos.success){
-            setTodos(props.todos.data)
-        }
-    }, [props.todos])
-
-
-    useEffect(()=>{
-        if(props.newItem.success){
+        if(newItem.success){
```

Comment:
Create container like this in separate files. Don't use **useDispatch** in component. you component is doing to much stuff here.

![image](https://user-images.githubusercontent.com/17162601/99366774-e5e67780-28da-11eb-9d78-a2bb8410943c.png)